### PR TITLE
Set zero offset for RYF

### PR DESCRIPTION
**Pull request by ezhilsabareesh8**
This PR addresses #16

### CODE REVIEW

**Comment by dougiesquire** [comment]
Thanks @ezhilsabareesh8. I think `generate_xml_datm.py` also needs a similar change

**Comment by ezhilsabareesh8** [comment]
> Thanks @ezhilsabareesh8. I think `generate_xml_datm.py` also needs a similar change

DATM is fine, both the RYF and IAF dataset starts at midnight. The offset is required only for few variables which starts at 1:30 hrs, which is already implemented in the code.

**Comment by ezhilsabareesh8** [comment]
> Thanks @ezhilsabareesh8. I think `generate_xml_datm.py` also needs a similar change

Fixed it in this commit ae25ac9

File: data_stream_xml_generation/generate_xml_datm.py [hunk 1]
```diff
@@ -84,9 +84,12 @@
...
-        SubElement(
+        if year_first == year_last:
+            SubElement(stream_info, "offset").text = "0"  #RYF starts at midnight
+        else:    
+            SubElement(
             stream_info, "offset"
-        ).text = "-5400"  # shift back 1.5hr to match RYF
+            ).text = "-5400"  # shift back 1.5hr to match RYF
     else:
         SubElement(stream_info, "offset").text = "0"
```
Comment (dougiesquire) [inline data_stream_xml_generation/generate_xml_datm.py:94]:
```python
    if (stream_name in [
        "CORE_IAF_JRA55do.PRSN",
        "CORE_IAF_JRA55do.PRRN",
        "CORE_IAF_JRA55do.LWDN",
        "CORE_IAF_JRA55do.SWDN",
    ]) and (year_first != year_last): 
        SubElement(
            stream_info, "offset"
        ).text = "-5400"  # shift back 1.5hr to match RYF
    else:
        SubElement(stream_info, "offset").text = "0"
```

**Review by dougiesquire — CHANGES_REQUESTED** [review]
Thanks @ezhilsabareesh8. One small change for clarity

File: data_stream_xml_generation/generate_xml_datm.py (hunk 1 above)
Comment (ezhilsabareesh8) [reply data_stream_xml_generation/generate_xml_datm.py:91]: Changed in this commit [9f28f27](https://github.com/COSIMA/om3-scripts/pull/17/commits/9f28f27ba493547adf0929f38fc08d2bc22f62b9)

File: data_stream_xml_generation/generate_xml_datm.py [hunk 2]
```diff
@@ -78,15 +78,15 @@
...
         "CORE_IAF_JRA55do.PRSN",
         "CORE_IAF_JRA55do.PRRN",
         "CORE_IAF_JRA55do.LWDN",
         "CORE_IAF_JRA55do.SWDN",
-    ]:
+    ]) and (year_first != year_last): 
         SubElement(
             stream_info, "offset"
-        ).text = "-5400"  # shift back 1.5hr to match RYF
+        ).text = "-5400"  # shift back 1.5hr to match RYF
```
Comment (dougiesquire) [inline data_stream_xml_generation/generate_xml_datm.py:89]:
```suggestion
        ).text = "-5400"  # shift back 1.5hr to match RYF
```

**Review by dougiesquire — CHANGES_REQUESTED** [review]

**Review by dougiesquire — APPROVED** [review]
Thanks @ezhilsabareesh8. LGTM. By the way, you can just accept the suggestions in my review, rather than having to go and make the commits yourself

**Pull request merged**